static/Functions.js

```diff
@@ -216,115 +214,124 @@ function processAttachments(isUpload, cb) {
-          if (response.isError) return cb(new Error('error uploading document: ' + response.message));
-
-          return cb(null, response);
+      }, function (err, response) {
+        if (err) return cb(err);
+
+        // in this case the document service might return a result that contains an error, so also need to check this specifically
+        // TODO: revisit api on document service after rewriting in Node.js.
+        // if there's an error it should send back a statusCode != 200 to indicate that
+        if (response.isError) return cb(new Error('error uploading document: ' + response.message));
+
+        return cb(null, response);
       });
     });
   });
 }
 
 function getFirstAttachmentHash(cb) {
 
-  return processAttachments(true, function(err, response) {
+  return processAttachments(true, function (err, response) {
     if (err) return cb(err);
-    console.log('got response', response);    
+    console.log('got response', response);
 
     if (!response.attachmentProcessingDetails || !response.attachmentProcessingDetails.length) {
       console.error('hash is not available');
       return cb(new Error('hash not available'));
     }
 
     var hash = response.attachmentProcessingDetails[0].hash;
-    return cb(null, { hash: hash });
+    return cb(null, {
+      hash: hash
+    });
 
   });
 }
 
 
 // TODO: revisit&rewrite this function
 function validateProof(event) {
-  return Office.context.mailbox.item.body.getAsync('text', {}, function(result) {
+  return Office.context.mailbox.item.body.getAsync('text', {}, function (result) {
     if (result.status === Office.AsyncResultStatus.Failed) {
       return showMessage(result.error, event);
     }
-    
+
     try {
       var body = result.value;
       if (body.search(beginProofString) === -1 || body.search(endProofString) === -1) {
-        return showMessage("No proofs to validate found in email", event);           
+        return showMessage("No proofs to validate found in email", event);
+      }
+
+      var proofsStep1Array = body.split(beginProofString);
+      var proofsStep2Array = proofsStep1Array[1].split(endProofString);
+
+      try {
+        var proofs = JSON.parse(proofsStep2Array[0]);
+      } catch (err) {
+        console.error('invalid json', proofsStep2Array[0]);
+        return showMessage("Invalid json", event);
       }
 
-      var proofs = body.split(beginProofString);
+      //var proofs = proofsObj.proofs;
+
+      if (!proofs.length) {
+        return showMessage("no proofs found", event);
+      }
 
       for (var i in proofs) {
-        if (proofs[i].search(endProofString) != -1) {
-          var proof = proofs[i].split(endProofString);
 
-          if (!proof.length) {
-            return showMessage("Unable to validate proof(s)", event); 
+        var trackingId = proofs[i].trackingId;
+        return getProof(trackingId, function (err, result) {
```

Comment:
change the name 'result' to 'getProofResult' so it is clearer later what this object is...
later on there's a line result.result.proofs, this is considered unreadable

Reply:
changed